Sprint_10

DenisBurenkov opened the pull request:
Создан SplashViewController и добавлена функциональноасть для перехода от контролера авторизации к основному контролеру приложения
Создан WKwebViwe
Созданы два Post запроса:
1 Запрос на авторизацию
2 Обработка ответа

DenisBurenkov committed:
Создан SplashViewController и добавлена функциональноасть для перехода от контролера авторизации к основному контролеру приложения
Создан WKwebViwe
Созданы два Post запроса:
1 Запрос на авторизацию
2 Обработка ответа

protos17 commented on Auth/AuthViewController.swift:
```diff
@@ -0,0 +1,34 @@
+}
+
+final class AuthViewController: UIViewController {
+    private let ShowWebViewSegueIdentifier = "ShowWebView"
```
Констаны принято писать с маленькой буквы

protos17 commented on Auth/OAuth2TokenStorage.swift:
```diff
@@ -0,0 +1,12 @@
+import Foundation
+
+class OAuth2TokenStorage {
```
Рекомендую реализовать синглтон(паттерн, при котором гарантируется, что будет создан только один экземпляр класса и предоставляет глобальную точку доступа к этому экземпляру). Мы должны в самом классе создать этот экземпляр, затем обращаться к этому экземпляру. В текущей реализации при каждом обращении создается новый экземпляр. Для этого нужно указать:

        static let shared = OAuth2TokenStorage()
        private init() {}

protos17 commented on Auth/OAuth2TokenStorage.swift:
```diff
@@ -0,0 +1,12 @@
+class OAuth2TokenStorage {
+    var token: String? {
+        get {
+            return UserDefaults.standard.string(forKey: "BearerToken")
```
"BearerToken" рекомендую вынести в приватную константу в классе и использовать ее, чтобы не ошибиться при написании

protos17 commented on Auth/AuthViewController.swift:
```diff
@@ -0,0 +1,34 @@
+        if segue.identifier == ShowWebViewSegueIdentifier {
+            guard
+                let webViewViewController = segue.destination as? WebViewViewController
+            else { fatalError("Failed to prepare for \(ShowWebViewSegueIdentifier)") }
```
Хочу отметить, что в продакшене не используется `fatalError`, кроме случаев, когда это необходимо, например, в банковском приложении при обнаружении джейлбрейка.

protos17 commented on Auth/OAuth2Service.swift:
```diff
@@ -0,0 +1,64 @@
+            case httpResponseErrorCode(Int)
+        }
+        
+        var components = URLComponents(string: "https://unsplash.com/oauth/token")
```
Рекомендую перенести в Constants

protos17 commented on Auth/OAuthTokenResponseBody.swift:
```diff
@@ -0,0 +1,8 @@
+import Foundation
+
+struct OAuthTokenResponseBody: Codable {
+    let access_token: String
+    let token_type: String
+    let scope: String
+    let created_at: Int
+}
```
Необходимо работать в camelCase стиле, а у тебя в snake_case. Чтобы исправить:

        struct OAuthTokenResponseBody: Codable {
           let accessToken: String
            ...
            private enum CodingKeys: String, CodingKey {
                case accessToken= "access_token"
                ...
            }
        }

protos17 commented on Auth/WebViewViewController.swift:
```diff
@@ -0,0 +1,99 @@
+import UIKit
+import WebKit
+
+fileprivate let UnsplashAuthorizeURLString = "https://unsplash.com/oauth/authorize"
```
Рекомендую перенести в Constants

protos17 commented on SplashViewController/SplashViewController.swift:
```diff
@@ -0,0 +1,70 @@
+import UIKit
+
+final class SplashViewController: UIViewController {
+    private let ShowAuthenticationScreenSegueIdentifier = "ShowAuthenticationScreen"
```
рекомендую написать с маленькой буквы

protos17 commented on SplashViewController/SplashViewController.swift:
```diff
@@ -0,0 +1,70 @@
+            guard let self else { return }
+            switch result {
+            case .success:
+                self.switchToTabBarController()
```
Необходимо добавить сохранение токена:

       self.oauth2TokenStorage.token = code

DenisBurenkov committed:
Исправленны мелкие ошибки в наименованиях
Добавлен патер сингилтон в класс OAuth2TokenStorage
Заменена обработка ошибок FatalError
Добавленна обработка наименований в OAuthTokenResponseBody

Исправлены все замечания ревьюера (очень надеюсь :) )
